src/Library/demos/Actions/main.blp

```diff
@@ -0,0 +1,234 @@
+
+              Label {
+                label: _("Scale");
+
+                styles ["heading"]
+              }
+
+              Box {
+                halign: center;
+                valign: center;
+
+                ToggleButton {
+                  action-name: "demo.scale";
+                  action-target: "'100%'";
+                  label: _("100%");
+                }
+
+                ToggleButton {
+                  action-name: "demo.scale";
+                  action-target: "'150%'";
+                  label: _("150%");
+                }
+
+                ToggleButton {
+                  action-name: "demo.scale";
+                  action-target: "'200%'";
+                  label: _("200%");
+                }
+
+                styles ["linked"]
+              }
+            }
+          }
+        }
+
+        Box {
+          orientation: vertical;
+          spacing: 6;
+
+          Label {
+            halign: start;
+            label: _("Specialized Actions");
+
+            styles ["title-4"]
+          }
+
+          Label {
+            halign: start;
+            label: _("A stateful action which is bound to a GObject property");
+
+            styles ["dim-label"]
+          }
+
+          Box {
+            orientation: vertical;
+            halign: center;
+            margin-top: 6;
+            margin-bottom: 12;
+            homogeneous: true;
+
+
+            Label text {
+              label: _("Text Justify");
+              halign: center;
+
+              styles ["heading"]
+            }
+
+            Box {
+              halign: center;
+              spacing: 6;
+
+              CheckButton {
+                label: _("Start");
+                action-name: "demo.text-justify";
+                action-target: "'start'";
+              }
+
+              CheckButton {
+                label: _("Center");
+                action-name: "demo.text-justify";
+                action-target: "'center'";
+              }
+
+              CheckButton {
+                label: _("End");
+                action-name: "demo.text-justify";
+                action-target: "'end'";
+              }
+            }
+          }
+        }
+      }
+
+      Box links {
+        halign: center;
+
+        LinkButton {
+          label: "GJS Guide";
+          uri: "https://gjs.guide/guides/gio/actions-and-menus.html";
+        }
+
+        LinkButton {
+          label: "GTK Documentation";
+          uri: "https://docs.gtk.org/gtk4/actions.html";
+        }
+      }
+    }
+  }
+}
+
+menu bookmarks_menu {
+  item {
+    label: _("Developer Documentation");
+    action: "demo.open-bookmarks";
+    target: _("Developer Documentation");
+  }
+
+  item {
+    label: _("Human Interface Guidelines");
+    action: "demo.open-bookmarks";
+    target: _("Human Interface Guidelines");
+  }
+
+  item {
+    label: _("GNOME Javascript");
+    action: "demo.open-bookmarks";
+    target: _("GNOME Javascript");
+  }
```

Comment:
These don't all have to be `app.open_uri`, but I think it's a good opportunity to demo `app` scoped actions. Maybe one could be `app.about`, and another the `demo.open-bookmarks` action?

```suggestion
  item {
    label: _("Developer Documentation");
    action: "app.open_uri";
    target: "https://developer.gnome.org";
  }

  item {
    label: _("Human Interface Guidelines");
    action: "app.open_uri";
    target: "https://developer.gnome.org/hig/";
  }

  item {
    label: _("GNOME Javascript");
    action: "app.open_uri";
    target: "https://gjs-docs.gnome.org";
  }
```

Reply:
So should I change them to ``app.open_uri`` or keep them ``demo.open-bookmarks``? I wanted to show an example of an action that only takes parameters and has no state. I'm not sure how to squeeze in a demo for ``app`` scoped actions though, it might be weird to have an "About App" section under the bookmarks?